sympy/stats/tests/test_mixture_rv.py

```diff
@@ -0,0 +1,112 @@
+from sympy import (S, symbols, sqrt, Abs, exp, pi, oo, erf, erfc, Integral,
+                Interval, Dummy, factorial, binomial, log, beta, lerchphi,
+                Piecewise, lowergamma, gamma)
+from sympy.stats import (P, E, density, sample, cdf, Normal, Laplace, Mixture,
+                NegativeBinomial, Poisson, Geometric, YuleSimon, Logarithmic,
+                Binomial, Hypergeometric, BetaBinomial)
+from sympy.stats.mixture_rv import MixtureDistribution, MixturePSpace
+from sympy.testing.pytest import raises, skip, ignore_warnings
+from sympy.external import import_module
+
+y, z = symbols('y z')
+def test_continuous_mixture():
+    N = Normal("N", 0, 1)
+    M = Normal('M', 1, 2)
+    Z = Laplace('L', 3, 1)
+    D = Mixture('D', [S(2)/10, S(5)/10, S(3)/10], [N, M, Z])
```

Comment:
Instead of creating a new Mixture RV, a user should be able to define a new distribution using, `MixtureDistribution` and then pass it on to custom RV API. `Mixture` can be removed.

Reply:
But for accessing P, E, etc. we need an RV. Considering that the MixtureDIstribution is created, then we need an RV which will take the MixtureDistrbution as an argument and then able to change the pspace to any of the Finite, Continuous or Discrete Pspaces upon calling of `compute_cdf`, `compute_density` etc., so we need  `Mixture` RV IMO. The change can be:
```py
>>> M = Mixture('M', [wt1, wt2, wt3, ...,], [dist1, dist2, dist3, ...,])
```

Reply:
I think, `Mixture` can be a function rather than a class. It can return a generic RV/ContinuousRV/DiscreteRV/FiniteRV depending on the type of PDF. 
P.S. I saw that ContinuousRV generates only RVs with continuous PDF but what about PDFs which are neither continuous nor discrete? What will be returned in that case?

Reply:
> I think, `Mixture` can be a function rather than a class

Do you mean to remove the `MixtureDistribution`/`MixturePSpace` class?


>  I saw that ContinuousRV generates only RVs with continuous PDF but what about PDFs which are neither continuous nor discrete?

Presently, I have considered the 3 cases of Continuous, Discrete, and Finite. Should an error be raised if the distribution is not out of these 3?

Reply:
> Should an error be raised if the distribution is not out of these 3?

No a generic random symbol can be returned having the custom PDF given by the user.

Reply:
> Do you mean to remove the `MixtureDistribution`/`MixturePSpace` class?

`Mixture` should be a constructor function returning generic RandomSymbol objects with the given mixture probability distribution.